modules/library/data/src/main/java/org/geotools/data/memory/MemoryEntry.java

```diff
@@ -60,7 +61,7 @@
     MemoryEntry( MemoryDataStore store, SimpleFeatureType schema){
```

Comment:
To enforce the synchronized functionality, should the `memory` map be made private?